FWCore/Concurrency/interface/WaitingTaskHolder.h

```diff
@@ -0,0 +1,84 @@
+#ifndef FWCore_Concurrency_WaitingTaskHolder_h
+#define FWCore_Concurrency_WaitingTaskHolder_h
+// -*- C++ -*-
+//
+// Package:     FWCore/Concurrency
+// Class  :     WaitingTaskHolder
+// 
+/**\class WaitingTaskHolder WaitingTaskHolder.h "WaitingTaskHolder.h"
+
+ Description: [one line class summary]
+
+ Usage:
+    <usage>
+
+*/
+//
+// Original Author:  FWCore
+//         Created:  Fri, 18 Nov 2016 20:30:42 GMT
+//
+
+// system include files
+#include <cassert>
+
+// user include files
+#include "FWCore/Concurrency/interface/WaitingTask.h"
+
+// forward declarations
+
+namespace edm {
+  class WaitingTaskHolder
+  {
+    
+  public:
+    WaitingTaskHolder():
+    m_task(nullptr) {}
+    
```

Comment:
You are correct that providing `= default` is equivalent (in behavior) to specifying `WaitingTaskHolder(){}`.  However, when in-class initialization is used, the member is still initialized to its user-provided, in-class default value, even if an initialization list is absent in the c'tor.  If an in-class, default-initialized member is also specified in the c'tor's initialization list, the c'tor initialization takes precedence.